From 2aaa6aa8c794a5c4b8e0ef2bf65c8819ad023dea Mon Sep 17 00:00:00 2001
From: Ido Yariv <ido@wizery.com>
Date: Tue, 13 Dec 2011 23:16:01 +0200
Subject: [PATCH 1/2] wl12xx: Decrease number of RX transactions
 *EXPERIMENTAL*

On weak platforms, it is crucial to keep the number of SDIO transactions
to a bare minimum. It is probably more important to keep CPU utilization
low, than to handle FW events asap.

In order to decrease the number of RX transactions, set the FW to only
wake up the host when it has at least 4 pending packets. In addition,
increase the relevant timeout, so the FW could actually reach this
threshold.
---
 drivers/net/wireless/wl12xx/main.c |    4 ++--
 1 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/drivers/net/wireless/wl12xx/main.c b/drivers/net/wireless/wl12xx/main.c
index 776e449..0c5a4e7 100644
--- a/drivers/net/wireless/wl12xx/main.c
+++ b/drivers/net/wireless/wl12xx/main.c
@@ -124,8 +124,8 @@ static struct conf_drv_settings default_conf = {
 		.rts_threshold               = IEEE80211_MAX_RTS_THRESHOLD,
 		.rx_cca_threshold            = 0,
 		.irq_blk_threshold           = 0xFFFF,
-		.irq_pkt_threshold           = 0,
-		.irq_timeout                 = 600,
+		.irq_pkt_threshold           = 4,
+		.irq_timeout                 = 1200,
 		.queue_type                  = CONF_RX_QUEUE_TYPE_LOW_PRIORITY,
 	},
 	.tx = {
-- 
1.7.7.4

